include/mysql/plugin.h

```diff
@@ -756,6 +756,8 @@ void thd_set_ha_data(MYSQL_THD thd, const struct handlerton *hton,
 */
 void thd_wakeup_subsequent_commits(MYSQL_THD thd, int wakeup_error);
 
+void thd_get_query_start_data(MYSQL_THD thd, char *buf);
```

Comment:
Maybe just put it once in `row0mysql.cc`? IMO it's better than several copies.

Reply:
Good point.